### PR TITLE
Vault test cluster helper refactorings, mostly audit related

### PR DESCRIPTION
**Pull request by ncabatoff**
Move some test helper stuff from the vault package to a new helper/testhelpers/corehelpers package.  Consolidate on a single "noop audit" implementation.

### CODE REVIEW

File: vault/audit_test.go [hunk 1]
```diff
@@ -406,12 +387,13 @@ func TestAuditBroker_LogRequest(t *testing.T) {
 		Request:  reqCopy,
 		OuterErr: reqErrs,
 	}
-	err = b.LogRequest(context.Background(), logInput, headersConf)
+	ctx := namespace.RootContext(context.Background())
```
Comment (stevendpclark) [inline vault/audit_test.go:390]: Is adding the `namespace.RootContext` wrapper around a background context here required, or just best practice?

**Review by stevendpclark — APPROVED** [review]

File: vault/audit_test.go (hunk 1 above)
Comment (ncabatoff) [reply vault/audit_test.go:390]: Required, we get a ErrNoNamespace otherwise.

Comment (stevendpclark) [reply vault/audit_test.go:390]: Due to merging of NoopAudit and now everyone gets a` n.formatter.AuditFormatWriter = &audit.JSONFormatWriter` I assume this became a requirement?

Comment (ncabatoff) [reply vault/audit_test.go:390]: Precisely.

**Pull request merged**